hudi-common/src/main/java/org/apache/hudi/expression/BindVisitor.java

```diff
@@ -0,0 +1,179 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.expression;
+
+import org.apache.hudi.internal.schema.Types;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BindVisitor implements ExpressionVisitor<Expression>  {
+
+  protected final Types.RecordType recordType;
+  protected final boolean caseSensitive;
+
+  public BindVisitor(Types.RecordType recordType, boolean caseSensitive) {
+    this.recordType = recordType;
+    this.caseSensitive = caseSensitive;
+  }
+
+  @Override
+  public Expression alwaysTrue() {
+    return Predicates.True.get();
+  }
+
+  @Override
+  public Expression alwaysFalse() {
+    return Predicates.False.get();
+  }
+
+  @Override
+  public Expression visitAnd(Predicates.And and) {
+    if (and.getLeft() instanceof Predicates.False
+        || and.getRight() instanceof Predicates.False) {
+      return alwaysFalse();
+    }
+
+    Expression left = and.getLeft().accept(this);
+    Expression right = and.getRight().accept(this);
+    if (left instanceof Predicates.False
+        || right instanceof Predicates.False) {
+      return alwaysFalse();
+    }
+
+    if (left instanceof Predicates.True
+        && right instanceof Predicates.True) {
+      return alwaysTrue();
+    }
+
+    if (left instanceof Predicates.True) {
+      return right;
+    }
+
+    if (right instanceof Predicates.True) {
+      return left;
+    }
+
+    return Predicates.and(left, right);
+  }
+
+  @Override
+  public Expression visitOr(Predicates.Or or) {
+    if (or.getLeft() instanceof Predicates.True
+        || or.getRight() instanceof Predicates.True) {
+      return alwaysTrue();
+    }
+
+    Expression left = or.getLeft().accept(this);
+    Expression right = or.getRight().accept(this);
+    if (left instanceof Predicates.True
+        || right instanceof Predicates.True) {
+      return alwaysTrue();
+    }
+
+    if (left instanceof Predicates.False
+        && right instanceof Predicates.False) {
+      return alwaysFalse();
+    }
+
+    if (left instanceof Predicates.False) {
+      return right;
+    }
+
+    if (right instanceof Predicates.False) {
+      return left;
+    }
+
+    return Predicates.or(left, right);
+  }
+
+  @Override
+  public Expression visitLiteral(Literal literal) {
+    return literal;
+  }
+
+  @Override
+  public Expression visitNameReference(NameReference attribute) {
+    // TODO Should consider caseSensitive?
```

Comment:
Related point: What is the general case sensitivity approach we have taken here for partition push down filters ?

Reply:
Now we get field value using case sensitive resolution, I'm thinking maybe we also need to support case insensitive.

Or maybe I should use case insensitive by default, since `spark.sql.caseSensitive` is default to false.